tests/kafkatest/services/kafka/kafka.py

```diff
@@ -94,7 +95,6 @@ def __init__(self, context, num_nodes, zk, security_protocol=SecurityConfig.PLAI
         self.zk_set_acl = False
         self.server_prop_overides = server_prop_overides
         self.log_level = "DEBUG"
-        self.num_nodes = num_nodes
```

Comment:
In ducktape 0.6.0, this assignment is not needed because the constructor of the Service superclass already does it.  In the next release of ducktape, this assignment will not be possible because num_nodes is being converted to a Python property.  Its value will be produced by a function rather than being an instance field.